### ISSUE TITLE
bugginess from major overhaul

### ISSUE THREAD

**Issue by lindsayplatt** [USGS-R/gsplot#344]
#342 made big changes, but passed tests

Here is a collection of things that no longer work as expected (and therefore need tests added). Details in the comments below.
- [ ] ~~no box around the plots~~ we know this is an issue, blocked by #350 
- [x] `las` defined in default.yaml not being used
- [x] oddly thick legend box
- [ ] ~~axis lines not extended to the actual limits~~ this is base behavior
- [ ] ~~legend not correctly spaced away from x axis label~~ this was an issue in previous versions
- [ ] ~~custom legend ability is gone~~ pulled out into #353

**Comment by lindsayplatt** [comment]
- [ ] ~~no box around the plots~~ we know this is an issue, blocked by #350 
- [x] `las` defined in default.yaml not being used
- [x] oddly thick legend box

Example plot that shows these issues:

``` r
MaumeeDV <- MaumeeDV

demoPlot <- gsplot() %>%
  points(y=c(3,1,2), x=1:3, xlim=c(0,NA),ylim=c(0,NA),
         col="blue", pch=18, legend.name="Points", xlab="Index") %>%
  lines(c(3,4,3), c(2,4,6), legend.name="Lines", ylab="Data") %>%
  abline(b=1, a=0, legend.name="1:1") %>%
  legend(location="topleft",title="Awesome!") %>%
  grid() %>%
  error_bar(x=1:3, y=c(3,1,2), y.high=c(0.5,0.25,1), y.low=0.1) %>%
  error_bar(x=1:3, y=c(3,1,2), x.low=.2, x.high=.2, col="red",lwd=3) %>%
  callouts(x=1, y=2.8, lwd=2, angle=250, labels="Weird data") %>%
  title("Graphing Fun")
demoPlot
```

![image](https://cloud.githubusercontent.com/assets/13220910/15189422/38aa3d98-1770-11e6-9db9-afa4a7a376dd.png)

**Comment by lindsayplatt** [comment]
- [ ] ~~axis lines not extended to the actual limits~~ that's actually normal (when a box is not drawn around, it looks like that, even in base: `plot(dates[[site]], Wtemp[[site]], col="orangered", type='l', frame.plot=F)`
- [ ] ~~legend not correctly spaced away from x axis label~~ this was an issue in previous versions

Example plot that shows these issues:

``` r
sites <- unique(MaumeeDV$site_no)
dates <- sapply(sites, function(x) MaumeeDV$Date[which(MaumeeDV$site_no==x)], USE.NAMES=TRUE)
flow <- sapply(sites, function(x) MaumeeDV$Flow[which(MaumeeDV$site_no==x)], USE.NAMES=TRUE)
pH <- sapply(sites, function(x) MaumeeDV$pH_Median[which(MaumeeDV$site_no==x)], USE.NAMES=TRUE)
Wtemp <- sapply(sites, function(x) MaumeeDV$Wtemp[which(MaumeeDV$site_no==x)], USE.NAMES=TRUE)

site <- '04193490'
demoPlot <- gsplot(mar=c(7.1, 4.1, 4.1, 4.1)) %>% 
  lines(dates[[site]], Wtemp[[site]], col="orangered", 
        legend.name="Water Temperature", ylab='Water Temperature (deg C)') %>%
  lines(dates[[site]], pH[[site]], col="seagreen", side=4, 
        legend.name="pH", ylab='pH (pH Units)') %>%
  title(main=paste("Site", site), xlab='time') %>% 
  legend(location="below")
demoPlot
```

![image](https://cloud.githubusercontent.com/assets/13220910/15189631/6eae3394-1771-11e6-80b4-6100644bb1fa.png)

**Comment by jordansread** [comment]
@lindsaycarr was the legend thing an issue with this PR or did that surface before it? The rest look good

**Comment by lindsayplatt** [comment]
@jread-usgs I don't remember. I might be noticing it more since there is not a box around the plot, so it stands out. 

By "the rest look good" do you mean "yes, those are issues"?

**Comment by jordansread** [comment]
yes, those are issues.

**Comment by jordansread** [comment]
Ok, just verified the legend thing is new per this PR. I think it is related to par

**Comment by jordansread** [comment]
For the legend box, `par("lwd")` is used because `box.lwd` is missing by default. `par("lwd")` is 2 in this case. Not sure why...must be par leakage from the `callouts` call?

**Comment by jordansread** [comment]
@lindsaycarr can you check out these again after #345 and close/modify this issue and/or create a new issue to capture remaining issues?

**Comment by lindsayplatt** [comment]
checked off the things that look good now. The axis limits are still strange, as well as the legend offset from the bottom of the plot

**Comment by ldecicco-USGS** [comment]
I think we lost our custom legend ability

``` r
gsplot() %>%
 points(1:4, 1:4, col=c("red","black","blue","green"), cex=1:4, pch=15:18) %>%
 legend(location = "topleft", pch=15:18,
       legend=c("a","b","c","d"),col="red", title = "Shape" ) 
```

What I'm not sure on is if it's captured in another issue.

**Comment by lindsayplatt** [comment]
@ldecicco-USGS also missing `las` from the yaml again - thought we had fixed that in #345 

### Before:

![image](https://cloud.githubusercontent.com/assets/13220910/16013064/aa83693a-3151-11e6-8884-6f48f078de70.png)

### After:

![image](https://cloud.githubusercontent.com/assets/13220910/16013086/b9f725e6-3151-11e6-845c-9cbd64391c22.png)

**Comment by ldecicco-USGS** [comment]
Actually, isn't that the right `las`? My #triggerWarning wasn't tingling.

As for framing (ie...whether or not to put a box around)...Maybe we could inject the plot.defaults at the first instance of gsplot(), and then only override them if the user does.

```
## Default S3 method:
plot(x, y = NULL, type = "p",  xlim = NULL, ylim = NULL,
     log = "", main = NULL, sub = NULL, xlab = NULL, ylab = NULL,
     ann = par("ann"), axes = TRUE, frame.plot = axes,
     panel.first = NULL, panel.last = NULL, asp = NA, ...)
```

That was my intent waaay back in the day. I can't figure out of that just got lost in the overhaul, or it's acting up.

**Comment by lindsayplatt** [comment]
^ yes, was the correct `las`. I think I was using the old gsplot version when I looked at that
